File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/context/roleplay/ChangeMapMessage.scala





















// Generated on 05/03/2013 10:34:42
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait ChangeMapMessage extends MessageDeserializer {
  val messageId = 221

  protected trait Message extends BaseMessage with Serializable {
    val messageId = ChangeMapMessage.messageId

    val mapId: Int

    def serialize(buf: Buffer) {
      buf.writeInt(mapId)
    }

  }

  protected def create(mapId: Int): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val mapId = buf.readInt()

    create(mapId)
  }

}

object ChangeMapMessage extends ChangeMapMessage {
  case class MessageImpl(mapId: Int) extends Message
  type Target = MessageImpl

  override def create(mapId: Int) = MessageImpl(mapId)

  def apply(mapId: Int) = create(mapId)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/ChangeMapMessage.java


// Generated on 05/08/2013 19:37:45
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class ChangeMapMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 221;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public int mapId;
    
    public ChangeMapMessage() { }
    
    public ChangeMapMessage(int mapId) {
        this.mapId = mapId;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeInt(mapId);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        mapId = buf.readInt();
        if (mapId < 0)
            throw new RuntimeException("Forbidden value on mapId = " + mapId + ", it doesn't respect the following condition : mapId < 0");
    }
    
}
